yoti-sdk-api/src/main/java/com/yoti/api/client/shareurl/policy/DynamicPolicy.java

```diff
@@ -10,34 +10,40 @@
 
 import com.yoti.api.attributes.AttributeConstants;
 import com.yoti.api.client.shareurl.constraint.Constraint;
+import com.yoti.api.client.shareurl.policy.profile.IdentityProfile;
 
 import com.fasterxml.jackson.annotation.JsonProperty;
 
 /**
  * Set of data required to request a sharing transaction
  */
-public class DynamicPolicy {
+public final class DynamicPolicy {
 
-    @JsonProperty("wanted")
+    @JsonProperty(Property.WANTED)
     private final Collection<WantedAttribute> wantedAttributes;
 
-    @JsonProperty("wanted_auth_types")
+    @JsonProperty(Property.WANTED_AUTH_TYPES)
     private final Set<Integer> wantedAuthTypes;
 
-    @JsonProperty("wanted_remember_me")
+    @JsonProperty(Property.WANTED_REMEMBER_ME)
     private final boolean wantedRememberMe;
 
-    @JsonProperty("wanted_remember_me_optional")
+    @JsonProperty(Property.WANTED_REMEMBER_ME_OPTIONAL)
     private final boolean wantedRememberMeOptional;
 
+    @JsonProperty(Property.IDENTITY_PROFILE_REQUIREMENTS)
+    private final IdentityProfile identityProfile;
+
     DynamicPolicy(Collection<WantedAttribute> wantedAttributes,
             Set<Integer> wantedAuthTypes,
             boolean wantedRememberMe,
-            boolean wantedRememberMeOptional) {
+            boolean wantedRememberMeOptional,
+            IdentityProfile identityProfile) {
```

Comment:
I notice no explicit "public" here on the constructor - so is this constructor visible to code from other packages? The new constuctor parameter is intentionally optional (with a default of null) in dotnet (where the constructor is public and is visible to consuming code). If this constuctor is visible to integrators then perhaps we should implement something similar here in Java - using overload(s) for the optional params? Certainly on dotnet side we have various Integrators using various classes directly (without their builders). If this is possible on the Java side shall we avoid breaking changes for such Integrators? Conversely, if this constructor is not visible to integrators code (and they must use the Builder to construct) then I don't see the need to change anything here.

Reply:
Appologies I am not yet fully setup with my Java IDE and yoti-java-sdk otherwise I would answer this question about visibility myself.

Reply:
> If this is possible on the Java side shall we avoid breaking changes for such Integrators?
I wouldn't say it's a breaking change when it comes to different languages. Not an expert in out SDKs but I don't think you can actually reuse anything from .NET to Java and viceversa

As general concept providing a Builder and restricting the constructor to package/private is a good practice to focus the validation in both Builder.setters (validity of the data provided) and Builder.build (presence of mandatory properties).

Reply:
> I don't know if this makes said constructor "private to the package" or visible to an Integrators consuming code?)

It does.

Reply:
Ok perfect. Thanks